src/hooks/useActiveDescendantListFocus/useActiveDescendantListFocusHooks.js

```diff
@@ -0,0 +1,242 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+import useKeyEvent from "../useKeyEvent";
+import useEventListener from "../useEventListener";
+import usePrevious from "../usePrevious";
+import { getNextSelectableIndex, getPreviousSelectableIndex } from "./useActiveDescendantListFocusHelpers";
+import { useListenFocusTriggers } from "hooks/useListenFocusTriggers";
+
+const ARROW_DIRECTIONS = {
+  UP: "ArrowUp",
+  DOWN: "ArrowDown",
+  RIGHT: "ArrowRight",
+  LEFT: "ArrowLeft"
+};
+
+const ENTER_KEY = "Enter";
+const SPACE_KEY = " ";
+
+export function useSupportArrowsKeyboardNavigation({
+  itemsCount,
+  focusedElementRef,
+  visualFocusItemIndex,
+  setVisualFocusItemIndex,
+  isHorizontalList,
+  isItemSelectable,
+  listenerOptions,
+  triggerByKeyboard,
+  setTriggerByKeyboard
+}) {
+  const nextArrow = isHorizontalList ? ARROW_DIRECTIONS.RIGHT : ARROW_DIRECTIONS.DOWN;
+  const backArrow = isHorizontalList ? ARROW_DIRECTIONS.LEFT : ARROW_DIRECTIONS.UP;
+
+  const onArrowKeyEvent = useCallback(
+    direction => {
+      // we desire to change the visual focus item only if the user pressed on the keyboard arrows keys while
+      // the focusedElementRef is naturally focus
+      if (document.activeElement !== focusedElementRef.current) {
+        return;
+      }
+
```

Comment:
will not be in this pr: we want to explain in docs what is the difference between normal/natural focus and visual focus